File: domain/src/main/java/com/chronos/mobile/domain/RemoteTimetableSource.kt
package com.chronos.mobile.domain

import com.chronos.mobile.core.model.OnlineSchedulePayload
import com.chronos.mobile.domain.model.AuthSnapshot
import com.chronos.mobile.domain.result.AppResult

interface RemoteTimetableSource {
    suspend fun fetchSchedule(
        authSnapshot: AuthSnapshot,
        weekNum: String? = null,
        yearTerm: String? = null,
    ): AppResult<OnlineSchedulePayload>
}

File: src/lib/domain/interfaces/remote-timetable-source.ts
import type { OnlineScheduleFetchResult } from '$lib/models/online-schedule';
import type { AuthSnapshot } from '$lib/models/auth';
import type { AppResult } from '../result/app-result';

export interface RemoteTimetableSource {
	fetchSchedule(
		authSnapshot: AuthSnapshot,
		weekNum?: string | null,
		yearTerm?: string | null
	): Promise<AppResult<OnlineScheduleFetchResult>>;
}
